### PR TITLE
add public accessor for the pool._timeout value

### LINKED ISSUE #3689: add public accessor for the pool._timeout value
**Migrated issue, originally created by Thomas Briles**

(edited by mike) in the form of a read-only Python descriptor with documentation.

### PR DESCRIPTION
**Pull request by irinade**
fixes : https://github.com/sqlalchemy/sqlalchemy/issues/3689

### CODE REVIEW

**Comment by zzzeek** [comment]
woop!  a real PR against a real issue!  thanks!  let's run it through the machine

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision 630d24cc74745c49dc62fe30ef4e82bfd88e1b16 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 630d24cc74745c49dc62fe30ef4e82bfd88e1b16: https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1057

**Review by sqla-tester — COMMENTED** [review]
**mike bayer** (zzzeek) wrote:

(2 comments)

a few changes.

File: lib/sqlalchemy/pool/impl.py [hunk 1]
```diff
@@ -182,6 +182,9 @@ def status(self):
     def size(self):
         return self._pool.maxsize
 
+    def timeout(self):
+        return self._pool._timeout
```
Comment (sqla-tester) [inline lib/sqlalchemy/pool/impl.py]: **mike bayer** (zzzeek) wrote:

_timeout is on "self", so need to make this "self._timeout".

File: test/engine/test_pool.py [hunk 2]
```diff
@@ -1113,6 +1113,12 @@ def status(pool):
         lazy_gc()
         assert not pool._refs
 
+    def test_timeout_accessor(self):
+        expected_timeout = 123
+        p = self._queuepool_fixture(
+            timeout=expected_timeout)
+        assert p.timeout() == expected_timeout
```
Comment (sqla-tester) [inline test/engine/test_pool.py]: **mike bayer** (zzzeek) wrote:

I know this is not necessarily consistent with other tests in this module but let's make it:

   eq_(p.timeout(), expected_timeout)

**Comment by zzzeek** [comment]
please reopen if you have plans to continue this contribution, thanks!

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

needs fixes, will reopen if contributor returns

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/1057 has been **abandoned**.  That means that at least for the moment I need to close this pull request. Sorry it didn't work out :(

**Pull request closed without merging**